app/services/fact_importer.rb

```diff
@@ -138,7 +140,7 @@ def update_facts
     time = Time.now.utc
     updated = 0
     db_facts_names = []
-    host.fact_values.joins(:fact_name).where('fact_names.type' => fact_name_class_name).reorder('').find_each do |record|
+    host.fact_values.includes([:fact_name]).joins(:fact_name).where(FACT_NAME_TYPE => fact_name_class_name).reorder('').find_each do |record|
```

Comment:
Not crucial, but we could probably drop a couple more object allocations here and make the code more readable at the same time: 
```suggestion
    host.fact_values.includes(:fact_name).joins(:fact_name).where(fact_names: {type: fact_name_class_name}).reorder(nil).find_each do |record|
```